lib/archethic_web/graphql_schema.ex

```diff
@@ -111,12 +111,24 @@ defmodule ArchethicWeb.GraphQLSchema do
       end)
     end
 
+    @desc """
+    List all the nodes registered in the network
+    """
     field :nodes, list_of(:node) do
       resolve(fn _, _ ->
         {:ok, Resolver.nodes()}
       end)
     end
 
+    @desc """
+    List the nearest endpoints nodes from the client's IP
+    """
+    field :nearest_endpoints, list_of(:node) do
```

Comment:
Add an additional GraphQL Schema Type on ArchethicWeb.GraphQLSchema.P2PType 
  object :node_ip do
    field(:ip, :string)
    field(:port, :integer)
  end
As we are not mapping other details it can return the above :node_ip struct